app/grandchallenge/archives/views.py

```diff
@@ -558,6 +564,115 @@ def get_serializer_class(self):
             return ArchiveItemSerializer
 
 
+class ArchiveItemJobListView(
+    LoginRequiredMixin,
+    ObjectPermissionRequiredMixin,
+    JobsList,
+):
+    model = Job
+    template_name = "archives/archive_item_job_list.html"
+    row_template = "algorithms/job_list_row.html"
+
+    search_fields = [
+        *JobsList.search_fields,
+        # Outputs
+        "outputs__interface__title",
+        "outputs__interface__description",
+        # Algorithm
+        "algorithm_image__algorithm__title",
+        "algorithm_image__algorithm__description",
+        "algorithm_image__comment",
+    ]
+
+    permission_required = (
+        f"{Archive._meta.app_label}.view_{ArchiveItem._meta.model_name}"
+    )
+    raise_exception = True
+
+    login_url = reverse_lazy("account_login")
+
+    @cached_property
+    def archive_item(self):
+        return get_object_or_404(ArchiveItem, pk=self.kwargs["pk"])
+
+    def get_permission_object(self):
+        return self.archive_item
+
+    def get_queryset(self):
+        input_civs = self.archive_item.values.all()
+        num_civs = len(input_civs)
+
+        jobs_subquery = Job.objects.filter(inputs__in=input_civs).distinct()
+
+        query_set = (
+            Job.objects.filter(pk__in=jobs_subquery)
```

Comment:
Why do the jobs_subquery separately? Isn't it more efficient (or at least not less efficient) to integrate this into the main query below?

Reply:
I've incorporated it with a subquery now. Which passes the tests, but I am unsure about performance improvements.